### PR TITLE
Fix Wallet shutdown in TriblerTunnel community

### PR DESCRIPTION
**Pull request by ichorid**
This patch adds calls to correctly tear down bandwidth wallet on shutdown of TriblerTunnel community.

Related to #3876

### CODE REVIEW

File: Tribler/Test/Community/Tunnel/test_community.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def create_node(self):
                                       working_directory=u":memory:")
         mock_ipv8.overlay.bandwidth_wallet = TrustchainWallet(overlay)
         mock_ipv8.overlay.dht_provider = MockDHTProvider(mock_ipv8.endpoint.wan_address)
+        self.private_nodes.append(mock_ipv8)
```
Comment (devos50) [inline Tribler/Test/Community/Tunnel/test_community.py]: What's the reason for this change?

**Comment by devos50** [comment]
Nice find 👍

File: Tribler/Test/Community/Tunnel/test_community.py (hunk 1 above)
Comment (ichorid) [reply Tribler/Test/Community/Tunnel/test_community.py]: `tearDown` in this test class unloads the `private_nodes`. `create_node` is used several times through this file, so each instance should be tracked to be properly unloaded. Before the patch it was only registered by hand at line 85. Now it is done automatically.

Comment (devos50) [reply Tribler/Test/Community/Tunnel/test_community.py]: Ok thanks for the explanation!

Comment (qstokkink) [reply Tribler/Test/Community/Tunnel/test_community.py]: But, create node is also used by `initialize`, so now all the nodes in `self.nodes` are also in `self.private_nodes`.

**Comment by ichorid** [comment]
This is the latest version of Trial branch. It's a bit cleaner.

File: Tribler/Test/Community/Tunnel/test_community.py [hunk 2]
```diff
@@ -38,24 +38,20 @@ class TestTriblerTunnelCommunity(TestBase):
...
-            node.unload()
-
     def create_node(self):
         mock_ipv8 = MockIPv8(u"curve25519", TriblerTunnelCommunity, socks_listen_ports=[],
                              exitnode_cache=join(mkdtemp(suffix="_tribler_test_cache"), 'cache.dat'))
         mock_ipv8.overlay.settings.max_circuits = 1
 
         # Load the TrustChain community
-        overlay = TrustChainCommunity(mock_ipv8.my_peer, mock_ipv8.endpoint, mock_ipv8.network,
+        mock_ipv8.trustchain = TrustChainCommunity(mock_ipv8.my_peer, mock_ipv8.endpoint, mock_ipv8.network,
```
Comment (qstokkink) [inline Tribler/Test/Community/Tunnel/test_community.py]: You can also set `create_trustchain=True` when initializing `MockIPv8`, it does the exact same thing.

**Comment by ichorid** [comment]
All this PRs really depend on each other. I rebased everything on top of the latest 'next' and updated IPv8 pointer.

File: Tribler/Test/Community/Tunnel/test_community.py (hunk 2 above)
Comment (qstokkink) [reply Tribler/Test/Community/Tunnel/test_community.py]: How was this resolved?

Comment (ichorid) [reply Tribler/Test/Community/Tunnel/test_community.py]: (my original answer got swallowed by GitHub smartness...)

If you try to add TrustChain object here by using `create_trustchain=True`, the test errors out:
``` python
Traceback (most recent call last):
  File "/tmp/tribler/Tribler/Test/Community/Tunnel/test_community.py", line 41, in setUp
    self.initialize(TriblerTunnelCommunity, 1)
  File "/tmp/tribler/Tribler/pyipv8/ipv8/test/base.py", line 37, in initialize
    self.nodes = [self.create_node(*args, **kwargs) for _ in range(node_count)]
  File "/tmp/tribler/Tribler/Test/Community/Tunnel/test_community.py", line 49, in create_node
    exitnode_cache=join(mkdtemp(suffix="_tribler_test_cache"), 'cache.dat'))
  File "/tmp/tribler/Tribler/pyipv8/ipv8/test/mocking/ipv8.py", line 40, in __init__
    self.overlay = overlay_class(self.my_peer, self.endpoint, self.network, *args, **kwargs)
  File "/tmp/tribler/Tribler/community/triblertunnel/community.py", line 45, in __init__
    super(TriblerTunnelCommunity, self).__init__(*args, **kwargs)
  File "/tmp/tribler/Tribler/pyipv8/ipv8/messaging/anonymization/hidden_services.py", line 25, in __init__
    super(HiddenTunnelCommunity, self).__init__(*args, **kwargs)
  File "/tmp/tribler/Tribler/pyipv8/ipv8/messaging/anonymization/community.py", line 154, in __init__
    super(TunnelCommunity, self).__init__(*args, **kwargs)
exceptions.TypeError: __init__() got an unexpected keyword argument 'trustchain'

```

Comment (qstokkink) [reply Tribler/Test/Community/Tunnel/test_community.py]: @ichorid you need `create_trustchain` instead of `trustchain`:

https://github.com/Tribler/py-ipv8/blob/1caefbf2c7654c9fb2301b5ad9243877f377ae2a/ipv8/test/mocking/ipv8.py#L15

Comment (qstokkink) [reply Tribler/Test/Community/Tunnel/test_community.py]: Aha, I see. Because of this line: https://github.com/Tribler/py-ipv8/blob/1caefbf2c7654c9fb2301b5ad9243877f377ae2a/ipv8/test/mocking/ipv8.py#L32
